synapse/cmds/cron.py

```diff
@@ -0,0 +1,598 @@
+        laststart = 'Never' if laststart is None else self._format_timestamp(laststart)
+        lastend = 'Never' if lastend is None else self._format_timestamp(lastend)
+        lastresult = cron.get('lastresult') or '<None>'
+
+        self.printf(f'iden:            {idenf}')
+        self.printf(f'user:            {user}')
+        self.printf(f'recurring:       {isrecur}')
+        self.printf(f'# starts:        {startcount}')
+        self.printf(f'last start time: {laststart}')
+        self.printf(f'last end time:   {lastend}')
+        self.printf(f'last result:     {lastresult}')
+        self.printf(f'query:           {query}')
+        if not recs:
+            self.printf(f'entries:         <None>')
+        else:
+            self.printf(f'entries:         {"incunit":10} {"incval":6} {"required"}')
+            for reqdict, incunit, incval in recs:
+                reqdict = reqdict or '<None>'
+                incunit = incunit or '<None>'
+                incval = incval or '<None>'
+                self.printf(f'                 {incunit:10} {incval:6} {reqdict}')
+
+    async def runCmdOpts(self, opts):
+        line = opts.get('line')
+        if line is None:
+            self.printf(self.__doc__)
+            return
+
+        core = self.getCmdItem()
+
+        parseinfo = await s_syntax.getRemoteParseInfo(core)
+        argv = s_syntax.Parser(parseinfo, line).stormcmd()
+        try:
+            opts = self._make_argparser().parse_args(argv)
+        except s_exc.ParserExit:
+            return
+
+        handlers = {
+            'add': self._handle_add,
+            'list': self._handle_list,
+            'mod': self._handle_mod,
+            'del': self._handle_del,
+            'stat': self._handle_stat
+        }
+        await handlers[opts.cmd](core, opts)
+
+class At(s_cli.Cmd):
+    '''
+Adds a non-recurring cron job that will execute a Storm query at one or more
+specified times.
+
+List/details/deleting cron jobs created with 'at' use the same commands as
+other cron jobs:  cron list/stat/del respectively.
+
+Syntax: at (time|+time delta)+ {query}
+
+Notes: This command accepts one or more time specifications followed by exactly
+one storm query in curly braces.  Each time specification may be in synapse
+time delta format (e.g + 1 day) or synapse time format (e.g.
+20501217030432101).  Seconds will be ignored, as cron jobs' granularity is
+limited to minutes.
+
+All times are interpreted as UTC.
+
+The other option for time specification is a relative time from now.  This
+consists of a plus sign, a positive integer, then one of 'minutes, hours,
+days'.
+
+Note that the record for a cron job is stored until explicitly deleted via
+"cron del".
+
+Examples:
+    # Run a storm query in 5 minutes
+    at +5 minutes {[inet:ipv4=1]}
+
+    # Run a storm query tomorrow and in a week
+    at +1 day +7 days {[inet:ipv4=1]}
+
+    # Run a query at the end of the year Zulu
+    at 20181231Z2359 15:30 {[inet:ipv4=1]}
```

Comment:
what's the 15:30 here? ( since 23:59 seems like it's specified )

Reply:
A mistake. TODO: rerun all the examples.